openfifth/aspen-discovery
# Cookie management updates

Test Plan:
For the test plan we will test the cookie preferences against Summon. 
Note: If you have set these preferences before, navigate to your account -> your preferences and check that your summon cookie preferences are turned off when you first carry out steps 13 onwards. 

1. Apply patch
2. Navigate to Primary Configuration -> Library Systems -> and select your current library
3. Scroll down to 'Data protection Regulations' and check the box next to 'Require Cookie Storage Consent'
4.  Navigate to the Administration Menu
5. You may get a Cookie Consent Popup, if you do, you can ignore it or click 'Essential Only', if you have set your cookies before you may not get this pop up
6. Navigate to System Administration -> Modules
7. Scroll down to Summon, click edit and check the enable box. You can leave the other boxes blank. 
8. In the left hand menu, you should now have a section named Summon.  
9. Click on Summon -> Settings then select Add New.
10. Use the information in Passbolt to set up Summon and save. 
11. Using the administration menu, select Primary Configuration -> Library Systems and select edit for your current library
12. Scroll down to Summon and in the Summon Settings dropdown, select the name you gave your settings in step 10, then save. 
13. Carry out a Summon Search by selecting 'Articles and Databases' in the dropdown at the top of the page.
14. Click on one of the record returned. 
15. Check the db table: user_summon_usage and notice that it does not update.
16. Navigate to your account then in the Account Settings, select 'Your Preferences'.
17. Scroll down to 'Cookies to Allow' and turn Summon on then save. 
18. Repeat steps 13 - 15, but notice that this time the db table has updated.
